fix: TypeScript-Fehler – SettingsModal presentationStyle + eslint-config-prettier types

## Beschreibung

`npx tsc --noEmit` schlägt mit zwei pre-existing Fehlern fehl. Diese blockieren aktuell keine CI-Jobs (CI nutzt `tsc --noEmit || true`), sollten aber behoben werden.

## Fehler

### 1. `src/components/SettingsModal.tsx:43` – ungültiger `presentationStyle`-Wert

```
error TS2769: No overload matches this call.
  Type '"none" | "pageSheet"' is not assignable to type '"pageSheet" | "fullScreen" | "formSheet" | "overFullScreen" | undefined'.
  Type '"none"' is not assignable to type '"pageSheet" | "fullScreen" | "formSheet" | "overFullScreen" | undefined'.
```

**Ursache:** `Platform.OS === 'ios' ? 'pageSheet' : 'none'` – `'none'` ist kein gültiger Wert für `Modal.presentationStyle` in den aktuellen React Native Typen. Auf Android/Web hat `presentationStyle` keine Wirkung, der Wert `'none'` ist aber typmäßig nicht erlaubt.

**Fix:** Den ternären Ausdruck so umschreiben, dass auf Nicht-iOS kein `presentationStyle` übergeben wird:
```tsx
presentationStyle={Platform.OS === 'ios' ? 'pageSheet' : undefined}
```

### 2. `eslint.config.js:7` – fehlende Typdeklaration für `eslint-config-prettier`

```
error TS7016: Could not find a declaration file for module 'eslint-config-prettier'.
  Try `npm i --save-dev @types/eslint-config-prettier` if it exists or add a new declaration (.d.ts) file containing `declare module 'eslint-config-prettier';`
```

**Ursache:** `eslint-config-prettier` liefert keine eigenen TypeScript-Typen mit, und `@types/eslint-config-prettier` existiert nicht auf npm. Die Datei `eslint.config.js` wird zwar von `tsc` erfasst, ist aber kein Teil des App-Codes.

**Fix:** `eslint.config.js` aus dem `tsconfig.json`-Include ausschließen (z. B. via `exclude`) oder eine lokale `eslint.config.d.ts` mit `declare module 'eslint-config-prettier';` anlegen.

## Auswirkung

- CI ist nicht betroffen (tsc läuft mit `|| true`)
- `tsc --noEmit` gibt Exit Code 2 zurück – verhindert sauberes TypeScript-Strict-Enforcement in Zukunft

--- a/__tests__/components/AddActivityModal.test.tsx
+++ b/__tests__/components/AddActivityModal.test.tsx
@@ -171,6 +171,37 @@ describe('AddActivityModal Component', () => {
     expect(root).toBeTruthy();
   });
 
+  it('accepts initialEndMonth prop without crashing', () => {
+    const { root } = render(
+      <AddActivityModal
+        visible={true}
+        plantName="Tomate"
+        initialMonth={2}
+        initialEndMonth={8}
+        onClose={mockOnClose}
+        onAdd={mockOnAdd}
+      />
+    );
+    expect(root).toBeTruthy();
+  });
+
+  it('uses initialEndMonth as endMonth when provided', () => {
+    const { getByText } = render(
+      <AddActivityModal
+        visible={true}
+        plantName="Tomate"
+        initialMonth={2}
+        initialEndMonth={8}
+        onClose={mockOnClose}
+        onAdd={mockOnAdd}
+      />
+    );
+    fireEvent.press(getByText('Hinzufügen'));
+    const [, startMonth, endMonth] = mockOnAdd.mock.calls[0];
+    expect(startMonth).toBe(2);
+    expect(endMonth).toBe(8);
+  });
+
   it('allows selecting a different activity type', () => {
     const { getAllByText } = render(
       <AddActivityModal visible={true} plantName="Tomate" onClose={mockOnClose} onAdd={mockOnAdd} />

--- a/__tests__/screens/CalendarScreen.test.tsx
+++ b/__tests__/screens/CalendarScreen.test.tsx
@@ -1,5 +1,5 @@
 import React from 'react';
-import { render, fireEvent } from '@testing-library/react-native';
+import { render, fireEvent, waitFor } from '@testing-library/react-native';
 import { CalendarScreen } from '../../src/screens/CalendarScreen';
 import { PlantProvider } from '../../src/contexts/PlantContext';
 import { LanguageProvider } from '../../src/contexts/LanguageContext';
@@ -20,10 +20,63 @@ jest.mock('../../src/hooks/useTheme', () => ({
   }),
 }));
 
+const testPlant = {
+  id: 'plant-1',
+  name: 'Tomate',
+  isDefault: false,
+  userId: null,
+  activities: [],
+  notes: '',
+  createdAt: 1748736000000,
+  updatedAt: 1748736000000,
+};
+
 jest.mock('@react-native-async-storage/async-storage', () => ({
-  getItem: jest.fn().mockResolvedValue(null),
+  getItem: jest
+    .fn()
+    .mockImplementation((key: string) =>
+      key === '@Pflanzkalender:plants'
+        ? Promise.resolve(JSON.stringify([testPlant]))
+        : Promise.resolve(null)
+    ),
   setItem: jest.fn().mockResolvedValue(undefined),
   removeItem: jest.fn().mockResolvedValue(undefined),
+}));
+
+// Capture callbacks passed to PlantRowsContainer so tests can trigger them
+let capturedOnPressMonth: ((plantId: string, monthIndex: number) => void) | null = null;
+let capturedOnPressMonthRange: ((plantId: string, start: number, end: number) => void) | null =
+  null;
+
+jest.mock('../../src/components/PlantRowsContainer', () => ({
+  PlantRowsContainer: (props: {
+    onPressMonth: (plantId: string, monthIndex: number) => void;
+    onPressMonthRange: (plantId: string, start: number, end: number) => void;
+  }) => {
+    capturedOnPressMonth = props.onPressMonth;
+    capturedOnPressMonthRange = props.onPressMonthRange;
+    return null;
+  },
+}));
+
+jest.mock('../../src/components/AddActivityModal', () => ({
+  AddActivityModal: (props: {
+    visible: boolean;
+    initialMonth?: number;
+    initialEndMonth?: number;
+    plantName?: string;
+    onClose?: () => void;
+    onAdd?: () => void;
+  }) => {
+    const { View, Text } = require('react-native');
+    if (!props.visible) return null;
+    return (
+      <View testID="add-activity-modal">
+        <Text testID="modal-initial-month">{props.initialMonth ?? 'none'}</Text>
+        <Text testID="modal-initial-end-month">{props.initialEndMonth ?? 'none'}</Text>
+      </View>
+    );
+  },
 }));
 
 const wrapper = ({ children }: { children: React.ReactNode }) => (
@@ -31,6 +84,11 @@ const wrapper = ({ children }: { children: React.ReactNode }) => (
     <PlantProvider>{children}</PlantProvider>
   </LanguageProvider>
 );
+
+beforeEach(() => {
+  capturedOnPressMonth = null;
+  capturedOnPressMonthRange = null;
+});
 
 describe('CalendarScreen – Rendering', () => {
   it('renders without crashing with required providers', () => {
@@ -68,7 +126,7 @@ describe('CalendarScreen – Zoom controls', () => {
     const { findByTestId, getByTestId } = render(<CalendarScreen />, { wrapper });
     await findByTestId('zoom-label');
     fireEvent.press(getByTestId('zoom-out'));
-    fireEvent.press(getByTestId('zoom-out')); // already at min, should not change
+    fireEvent.press(getByTestId('zoom-out'));
     expect(getByTestId('zoom-label').props.children).toBe('75%');
   });
 
@@ -76,7 +134,7 @@ describe('CalendarScreen – Zoom controls', () => {
     const { findByTestId, getByTestId } = render(<CalendarScreen />, { wrapper });
     await findByTestId('zoom-label');
     fireEvent.press(getByTestId('zoom-in'));
-    fireEvent.press(getByTestId('zoom-in')); // already at max, should not change
+    fireEvent.press(getByTestId('zoom-in'));
     expect(getByTestId('zoom-label').props.children).toBe('133%');
   });
 
@@ -96,31 +154,78 @@ describe('CalendarScreen – Zoom controls', () => {
 });
 
 describe('CalendarScreen – Drag-to-create range selection', () => {
-  it('opens AddActivityModal with initialMonth and initialEndMonth when month range is selected', async () => {
-    const { findByTestId, queryByTestId } = render(<CalendarScreen />, { wrapper });
-    // Modal is initially hidden
-    let modal = queryByTestId('add-activity-modal');
-    expect(modal).toBeFalsy();
-    // Simulate range selection on a row (would be triggered by PlantRow's drag)
-    // For this test, we verify the modal would receive the correct props by checking visibility
-    // In a full integration test, PlantRow would call onPressMonthRange with plant ID and indices
-    // The CalendarScreen would then set selectedMonth, selectedEndMonth, and show the modal
-    // Since we can't easily trigger PlantRow's drag in unit test, we verify the logic is wired
-    expect(typeof CalendarScreen).toBe('function');
+  // In the test environment Dimensions returns portrait (750x1334),
+  // so CalendarScreen maps indices via portrait formula: startHalf = idx*4, endHalf = min(idx*4+3, 23)
+  // AddActivityModal only renders when selectedPlant is set, so we must wait for plant loading first.
+
+  async function waitForReady() {
+    // Wait until PlantRowsContainer has been rendered (callbacks captured) and
+    // PlantContext has finished loading from AsyncStorage (getItem resolves async).
+    await waitFor(() => {
+      expect(capturedOnPressMonthRange).not.toBeNull();
+      expect(capturedOnPressMonth).not.toBeNull();
+    });
+    // Flush all pending microtasks / state updates from the async AsyncStorage load
+    await new Promise((r) => setImmediate(r));
+    await new Promise((r) => setImmediate(r));
+  }
+
+  it('opens AddActivityModal when onPressMonthRange is triggered', async () => {
+    const { queryByTestId } = render(<CalendarScreen />, { wrapper });
+    expect(queryByTestId('add-activity-modal')).toBeFalsy();
+
+    await waitForReady();
+    capturedOnPressMonthRange!('plant-1', 2, 4);
+
+    await waitFor(() => expect(queryByTestId('add-activity-modal')).toBeTruthy());
   });
 
-  it('correctly maps portrait mode indices to half-months', () => {
-    // Portrait: 4 cells per month, so slot 0 = hm 0, slot 1 = hm 4, slot 2 = hm 8, etc.
-    // startIdx=0, endIdx=2 (portrait) should give startHalf=0, endHalf=11 (2*4+3 clamped to 23)
-    // This test verifies the mapping formula in handlePressMonthRange
-    expect(0 * 4).toBe(0); // startIdx=0 → startHalf=0
-    expect(Math.min(2 * 4 + 3, 23)).toBe(11); // endIdx=2 → endHalf=11
+  it('maps portrait indices correctly: startHalf=idx*4, endHalf=min(idx*4+3,23)', async () => {
+    const { queryByTestId } = render(<CalendarScreen />, { wrapper });
+
+    await waitForReady();
+    // idx 2..4 → startHalf=8, endHalf=19
+    capturedOnPressMonthRange!('plant-1', 2, 4);
+
+    await waitFor(() => {
+      expect(queryByTestId('modal-initial-month')?.props.children).toBe(8);
+      expect(queryByTestId('modal-initial-end-month')?.props.children).toBe(19);
+    });
   });
 
-  it('correctly maps landscape mode indices to half-months', () => {
-    // Landscape: indices map 1:1 to half-months
-    // startIdx=5, endIdx=8 should give startHalf=5, endHalf=8
-    expect(5).toBe(5); // startIdx=5 → startHalf=5
-    expect(8).toBe(8); // endIdx=8 → endHalf=8
+  it('clamps endHalf to 23 for last portrait cell', async () => {
+    const { queryByTestId } = render(<CalendarScreen />, { wrapper });
+
+    await waitForReady();
+    // idx 5..5 → startHalf=20, endHalf=min(23,23)=23
+    capturedOnPressMonthRange!('plant-1', 5, 5);
+
+    await waitFor(() => {
+      expect(queryByTestId('modal-initial-month')?.props.children).toBe(20);
+      expect(queryByTestId('modal-initial-end-month')?.props.children).toBe(23);
+    });
+  });
+
+  it('opens AddActivityModal when a single month is pressed', async () => {
+    const { queryByTestId } = render(<CalendarScreen />, { wrapper });
+    expect(queryByTestId('add-activity-modal')).toBeFalsy();
+
+    await waitForReady();
+    capturedOnPressMonth!('plant-1', 1);
+
+    await waitFor(() => expect(queryByTestId('add-activity-modal')).toBeTruthy());
+  });
+
+  it('passes initialMonth (portrait: idx*4) without initialEndMonth for single-month press', async () => {
+    const { queryByTestId } = render(<CalendarScreen />, { wrapper });
+
+    await waitForReady();
+    // idx 1 → selectedMonth = 1*4 = 4, selectedEndMonth = undefined
+    capturedOnPressMonth!('plant-1', 1);
+
+    await waitFor(() => {
+      expect(queryByTestId('modal-initial-month')?.props.children).toBe(4);
+      expect(queryByTestId('modal-initial-end-month')?.props.children).toBe('none');
+    });
   });
 });

--- a/__tests__/screens/CalendarScreen.test.tsx
+++ b/__tests__/screens/CalendarScreen.test.tsx
@@ -94,3 +94,33 @@ describe('CalendarScreen – Zoom controls', () => {
     expect(btn.props.accessibilityLabel).toBe('Zoom in');
   });
 });
+
+describe('CalendarScreen – Drag-to-create range selection', () => {
+  it('opens AddActivityModal with initialMonth and initialEndMonth when month range is selected', async () => {
+    const { findByTestId, queryByTestId } = render(<CalendarScreen />, { wrapper });
+    // Modal is initially hidden
+    let modal = queryByTestId('add-activity-modal');
+    expect(modal).toBeFalsy();
+    // Simulate range selection on a row (would be triggered by PlantRow's drag)
+    // For this test, we verify the modal would receive the correct props by checking visibility
+    // In a full integration test, PlantRow would call onPressMonthRange with plant ID and indices
+    // The CalendarScreen would then set selectedMonth, selectedEndMonth, and show the modal
+    // Since we can't easily trigger PlantRow's drag in unit test, we verify the logic is wired
+    expect(typeof CalendarScreen).toBe('function');
+  });
+
+  it('correctly maps portrait mode indices to half-months', () => {
+    // Portrait: 4 cells per month, so slot 0 = hm 0, slot 1 = hm 4, slot 2 = hm 8, etc.
+    // startIdx=0, endIdx=2 (portrait) should give startHalf=0, endHalf=11 (2*4+3 clamped to 23)
+    // This test verifies the mapping formula in handlePressMonthRange
+    expect(0 * 4).toBe(0); // startIdx=0 → startHalf=0
+    expect(Math.min(2 * 4 + 3, 23)).toBe(11); // endIdx=2 → endHalf=11
+  });
+
+  it('correctly maps landscape mode indices to half-months', () => {
+    // Landscape: indices map 1:1 to half-months
+    // startIdx=5, endIdx=8 should give startHalf=5, endHalf=8
+    expect(5).toBe(5); // startIdx=5 → startHalf=5
+    expect(8).toBe(8); // endIdx=8 → endHalf=8
+  });
+});

--- a/__tests__/screens/SettingsScreen.test.tsx
+++ b/__tests__/screens/SettingsScreen.test.tsx
@@ -6,6 +6,14 @@ import { LanguageProvider } from '../../src/contexts/LanguageContext';
 import AsyncStorage from '@react-native-async-storage/async-storage';
 
 const mockSetThemeMode = jest.fn();
+const mockRouterReplace = jest.fn();
+
+jest.mock('expo-router', () => ({
+  useRouter: () => ({
+    replace: mockRouterReplace,
+    back: jest.fn(),
+  }),
+}));
 
 jest.mock('../../src/hooks/useTheme', () => ({
   useTheme: () => ({
@@ -143,6 +151,13 @@ describe('SettingsScreen', () => {
     const { getByText } = renderWithProviders(<SettingsScreen />);
     expect(getByText(/Feedback|feedback/i)).toBeTruthy();
     expect(getByText('Ko-fi')).toBeTruthy();
+  });
+
+  it('navigates to "/" (not "/(tabs)") when Schließen/Close is pressed', () => {
+    const { getByText } = renderWithProviders(<SettingsScreen />);
+    fireEvent.press(getByText(/Schließen|Close/));
+    expect(mockRouterReplace).toHaveBeenCalledWith('/');
+    expect(mockRouterReplace).not.toHaveBeenCalledWith('/(tabs)');
   });
 
   it('is a valid React component', () => {

--- a/src/components/AddActivityModal.tsx
+++ b/src/components/AddActivityModal.tsx
@@ -1,4 +1,4 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import {
   View,
   Text,
@@ -16,6 +16,7 @@ interface AddActivityModalProps {
   visible: boolean;
   plantName: string;
   initialMonth?: number;
+  initialEndMonth?: number;
   onClose: () => void;
   onAdd: (type: string, startMonth: number, endMonth: number, color: string, label: string) => void;
 }
@@ -24,6 +25,7 @@ export const AddActivityModal: React.FC<AddActivityModalProps> = ({
   visible,
   plantName,
   initialMonth = 0,
+  initialEndMonth,
   onClose,
   onAdd,
 }) => {
@@ -31,9 +33,19 @@ export const AddActivityModal: React.FC<AddActivityModalProps> = ({
   const { t } = useLanguage();
   const [selectedType, setSelectedType] = useState(ACTIVITY_TYPES[0].type);
   const [startMonth, setStartMonth] = useState(initialMonth);
-  const [endMonth, setEndMonth] = useState(initialMonth);
+  const [endMonth, setEndMonth] = useState(initialEndMonth ?? initialMonth);
   const [customLabel, setCustomLabel] = useState('');
   const [rangeError, setRangeError] = useState('');
+
+  useEffect(() => {
+    if (visible) {
+      setStartMonth(initialMonth);
+      setEndMonth(initialEndMonth ?? initialMonth);
+      setSelectedType(ACTIVITY_TYPES[0].type);
+      setCustomLabel('');
+      setRangeError('');
+    }
+  }, [visible, initialMonth, initialEndMonth]);
 
   const selectedActivityType = ACTIVITY_TYPES.find((at) => at.type === selectedType);
   const typeLabel = (type: string) => t(`activity.type.${type}`) as string;
@@ -46,22 +58,11 @@ export const AddActivityModal: React.FC<AddActivityModalProps> = ({
     }
     if (selectedActivityType) {
       onAdd(selectedType, startMonth, endMonth, selectedActivityType.color, label);
-      // Reset
-      setSelectedType(ACTIVITY_TYPES[0].type);
-      setStartMonth(initialMonth);
-      setEndMonth(initialMonth);
-      setCustomLabel('');
-      setRangeError('');
       onClose();
     }
   };
 
   const handleCancel = () => {
-    setSelectedType(ACTIVITY_TYPES[0].type);
-    setStartMonth(initialMonth);
-    setEndMonth(initialMonth);
-    setCustomLabel('');
-    setRangeError('');
     onClose();
   };
 

--- a/src/components/PlantRow.tsx
+++ b/src/components/PlantRow.tsx
@@ -1,15 +1,22 @@
-import React, { useMemo, useState } from 'react';
-import { View, Text, StyleSheet, TouchableOpacity, TextInput } from 'react-native';
+import React, { useMemo, useState, useRef, useEffect, useCallback } from 'react';
+import { View, Text, StyleSheet, TouchableOpacity, TextInput, Platform } from 'react-native';
 import { Plant } from '../types';
 import { ActivityBar } from './ActivityBar';
 import { useTheme } from '../hooks/useTheme';
 import { usePlants } from '../contexts/PlantContext';
 import { calculateActivityRows } from '../utils/activityLayout';
 
+interface MonthCellWebProps {
+  onMouseDown?: (e: React.MouseEvent) => void;
+  onMouseEnter?: () => void;
+  style?: React.ComponentProps<typeof View>['style'];
+}
+
 interface PlantRowProps {
   plant: Plant;
   onPressActivity?: (activityId: string) => void;
   onPressMonth?: (monthIndex: number) => void;
+  onPressMonthRange?: (startMonth: number, endMonth: number) => void;
   onPressPlant?: () => void;
   totalMonths?: number; // Anzahl der sichtbaren Monate
   currentHalfMonth?: number; // Aktueller Halbmonat (0-23)
@@ -21,6 +28,7 @@ export const PlantRow: React.FC<PlantRowProps> = ({
   plant,
   onPressActivity,
   onPressMonth,
+  onPressMonthRange,
   onPressPlant: _onPressPlant,
   totalMonths = 24,
   currentHalfMonth,
@@ -31,6 +39,52 @@ export const PlantRow: React.FC<PlantRowProps> = ({
   const { updatePlant } = usePlants();
   const [isEditingNotes, setIsEditingNotes] = useState(false);
   const [notesText, setNotesText] = useState(plant.notes);
+
+  // Drag-to-create state (web only)
+  const [dragPreview, setDragPreview] = useState<{ startMonth: number; endMonth: number } | null>(
+    null
+  );
+  const dragStateRef = useRef<{ startMonth: number; endMonth: number } | null>(null);
+  const onPressMonthRef = useRef(onPressMonth);
+  const onPressMonthRangeRef = useRef(onPressMonthRange);
+  useEffect(() => {
+    onPressMonthRef.current = onPressMonth;
+  }, [onPressMonth]);
+  useEffect(() => {
+    onPressMonthRangeRef.current = onPressMonthRange;
+  }, [onPressMonthRange]);
+
+  // Global mouseup listener to finalize drag (web only, registered only during active drag)
+  useEffect(() => {
+    if (Platform.OS !== 'web' || !dragStateRef.current) return;
+    const handleMouseUp = () => {
+      const ds = dragStateRef.current;
+      if (!ds) return;
+      const start = Math.min(ds.startMonth, ds.endMonth);
+      const end = Math.max(ds.startMonth, ds.endMonth);
+      if (start !== end) {
+        onPressMonthRangeRef.current?.(start, end);
+      } else {
+        onPressMonthRef.current?.(start);
+      }
+      dragStateRef.current = null;
+      setDragPreview(null);
+    };
+    window.addEventListener('mouseup', handleMouseUp); // platform-safe
+    return () => window.removeEventListener('mouseup', handleMouseUp); // platform-safe
+  }, [dragPreview]);
+
+  const handleCellMouseDown = useCallback((e: React.MouseEvent, monthIndex: number) => {
+    e.preventDefault();
+    dragStateRef.current = { startMonth: monthIndex, endMonth: monthIndex };
+    setDragPreview({ startMonth: monthIndex, endMonth: monthIndex });
+  }, []);
+
+  const handleCellMouseEnter = useCallback((monthIndex: number) => {
+    if (!dragStateRef.current) return;
+    dragStateRef.current.endMonth = monthIndex;
+    setDragPreview({ startMonth: dragStateRef.current.startMonth, endMonth: monthIndex });
+  }, []);
 
   const months = Array.from({ length: totalMonths }, (_, i) => i);
   const gridWidth = totalMonths * cellWidth;
@@ -58,31 +112,66 @@ export const PlantRow: React.FC<PlantRowProps> = ({
     handleNotesSubmit();
   };
 
+  // Compute drag preview bar dimensions
+  const dragPreviewBar =
+    dragPreview !== null
+      ? (() => {
+          const start = Math.min(dragPreview.startMonth, dragPreview.endMonth);
+          const end = Math.max(dragPreview.startMonth, dragPreview.endMonth);
+          return {
+            left: `${(start / totalMonths) * 100}%`,
+            width: `${((end - start + 1) / totalMonths) * 100}%`,
+          };
+        })()
+      : null;
+
   return (
     <View style={[styles.row, { minHeight }]}>
-      <View style={[styles.monthsContainer, { width: gridWidth }]}>
+      <View
+        style={[
+          styles.monthsContainer,
+          { width: gridWidth },
+          Platform.OS === 'web'
+            ? ({ userSelect: 'none' } as React.ComponentProps<typeof View>['style'])
+            : undefined,
+        ]}
+      >
         {months.map((monthIndex) => {
           const absoluteMonthIndex = monthIndex + monthOffset;
           const isCurrentHalfMonth =
             currentHalfMonth !== undefined && absoluteMonthIndex === currentHalfMonth;
 
+          const baseStyle = [
+            styles.monthCell,
+            {
+              width: cellWidth,
+              borderColor: theme.border,
+              backgroundColor: isCurrentHalfMonth ? theme.border : 'transparent',
+            },
+          ];
+
+          if (Platform.OS === 'web') {
+            const webProps: MonthCellWebProps = {
+              onMouseDown: (e: React.MouseEvent) => handleCellMouseDown(e, monthIndex),
+              onMouseEnter: () => handleCellMouseEnter(monthIndex),
+              style: [
+                ...baseStyle,
+                { cursor: 'crosshair' } as React.ComponentProps<typeof View>['style'],
+              ],
+            };
+            return <View key={monthIndex} {...(webProps as React.ComponentProps<typeof View>)} />;
+          }
+
           return (
             <TouchableOpacity
               key={monthIndex}
-              style={[
-                styles.monthCell,
-                {
-                  width: cellWidth,
-                  borderColor: theme.border,
-                  backgroundColor: isCurrentHalfMonth ? theme.border : 'transparent',
-                },
-              ]}
+              style={baseStyle}
               onPress={() => onPressMonth?.(monthIndex)}
             />
           );
         })}
 
-        <View style={styles.activitiesLayer}>
+        <View style={styles.activitiesLayer} pointerEvents="box-none">
           {activitiesWithRows.map((activity) => (
             <View key={activity.id} style={[styles.activityContainer, { top: activity.row * 28 }]}>
               <ActivityBar
@@ -92,6 +181,29 @@ export const PlantRow: React.FC<PlantRowProps> = ({
               />
             </View>
           ))}
+          {/* Drag preview bar */}
+          {dragPreviewBar && (
+            <View
+              pointerEvents="none"
+              style={[
+                styles.dragPreviewBar,
+                {
+                  left: dragPreviewBar.left,
+                  width: dragPreviewBar.width,
+                  backgroundColor: theme.primary + '55',
+                  borderColor: theme.primary,
+                },
+              ]}
+            />
+          )}
+          {/* Empty row hint */}
+          {plant.activities.length === 0 && !dragPreview && Platform.OS === 'web' && (
+            <View style={styles.emptyHint} pointerEvents="none">
+              <Text style={[styles.emptyHintText, { color: theme.textSecondary }]}>
+                {'← Klicken oder ziehen →'}
+              </Text>
+            </View>
+          )}
         </View>
       </View>
 
@@ -162,5 +274,26 @@ const styles = StyleSheet.create({
     left: 0,
     right: 0,
     height: 24,
+  },
+  dragPreviewBar: {
+    position: 'absolute',
+    top: 0,
+    height: 24,
+    borderRadius: 4,
+    borderWidth: 1,
+    borderStyle: 'dashed',
+  },
+  emptyHint: {
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    right: 0,
+    bottom: 0,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  emptyHintText: {
+    fontSize: 10,
+    opacity: 0.5,
   },
 });

--- a/src/components/PlantRow.tsx
+++ b/src/components/PlantRow.tsx
@@ -52,6 +52,7 @@ export const PlantRow: React.FC<PlantRowProps> = ({
   const [dragPreview, setDragPreview] = useState<{ startMonth: number; endMonth: number } | null>(
     null
   );
+  const [isDragging, setIsDragging] = useState(false);
   const dragStateRef = useRef<{ startMonth: number; endMonth: number } | null>(null);
   const onPressMonthRef = useRef(onPressMonth);
   const onPressMonthRangeRef = useRef(onPressMonthRange);
@@ -62,36 +63,39 @@ export const PlantRow: React.FC<PlantRowProps> = ({
     onPressMonthRangeRef.current = onPressMonthRange;
   }, [onPressMonthRange]);
 
-  // Global mouseup listener to finalize drag (web only, registered only during active drag)
+  // Global mouseup listener – registered once when drag starts, removed when it ends
   useEffect(() => {
-    if (Platform.OS !== 'web' || !dragStateRef.current) return;
+    if (Platform.OS !== 'web' || !isDragging) return;
     const handleMouseUp = () => {
       const ds = dragStateRef.current;
-      if (!ds) return;
-      const start = Math.min(ds.startMonth, ds.endMonth);
-      const end = Math.max(ds.startMonth, ds.endMonth);
-      if (start !== end) {
-        onPressMonthRangeRef.current?.(start, end);
-      } else {
-        onPressMonthRef.current?.(start);
+      if (ds) {
+        const start = Math.min(ds.startMonth, ds.endMonth);
+        const end = Math.max(ds.startMonth, ds.endMonth);
+        if (start !== end) {
+          onPressMonthRangeRef.current?.(start, end);
+        } else {
+          onPressMonthRef.current?.(start);
+        }
       }
       dragStateRef.current = null;
       setDragPreview(null);
+      setIsDragging(false);
     };
     window.addEventListener('mouseup', handleMouseUp); // platform-safe
     return () => window.removeEventListener('mouseup', handleMouseUp); // platform-safe
-  }, [dragPreview]);
+  }, [isDragging]);
 
-  const handleCellMouseDown = useCallback((e: React.MouseEvent, monthIndex: number) => {
+  const handleCellMouseDown = useCallback((e: React.MouseEvent, absoluteIndex: number) => {
     e.preventDefault();
-    dragStateRef.current = { startMonth: monthIndex, endMonth: monthIndex };
-    setDragPreview({ startMonth: monthIndex, endMonth: monthIndex });
+    dragStateRef.current = { startMonth: absoluteIndex, endMonth: absoluteIndex };
+    setDragPreview({ startMonth: absoluteIndex, endMonth: absoluteIndex });
+    setIsDragging(true);
   }, []);
 
-  const handleCellMouseEnter = useCallback((monthIndex: number) => {
+  const handleCellMouseEnter = useCallback((absoluteIndex: number) => {
     if (!dragStateRef.current) return;
-    dragStateRef.current.endMonth = monthIndex;
-    setDragPreview({ startMonth: dragStateRef.current.startMonth, endMonth: monthIndex });
+    dragStateRef.current.endMonth = absoluteIndex;
+    setDragPreview({ startMonth: dragStateRef.current.startMonth, endMonth: absoluteIndex });
   }, []);
 
   const months = Array.from({ length: totalMonths }, (_, i) => i);
@@ -160,21 +164,28 @@ export const PlantRow: React.FC<PlantRowProps> = ({
 
           if (Platform.OS === 'web') {
             const webProps: MonthCellWebProps = {
-              onMouseDown: (e: React.MouseEvent) => handleCellMouseDown(e, monthIndex),
-              onMouseEnter: () => handleCellMouseEnter(monthIndex),
+              onMouseDown: (e: React.MouseEvent) => handleCellMouseDown(e, absoluteMonthIndex),
+              onMouseEnter: () => handleCellMouseEnter(absoluteMonthIndex),
               style: [
                 ...baseStyle,
                 { cursor: 'crosshair' } as unknown as React.ComponentProps<typeof View>['style'],
               ],
             };
-            return <View key={monthIndex} {...(webProps as React.ComponentProps<typeof View>)} />;
+            return (
+              <View
+                key={monthIndex}
+                accessibilityRole="button"
+                accessibilityLabel={`Monat ${absoluteMonthIndex + 1}`}
+                {...(webProps as React.ComponentProps<typeof View>)}
+              />
+            );
           }
 
           return (
             <TouchableOpacity
               key={monthIndex}
               style={baseStyle}
-              onPress={() => onPressMonth?.(monthIndex)}
+              onPress={() => onPressMonth?.(absoluteMonthIndex)}
             />
           );
         })}

--- a/src/components/PlantRow.tsx
+++ b/src/components/PlantRow.tsx
@@ -1,5 +1,13 @@
 import React, { useMemo, useState, useRef, useEffect, useCallback } from 'react';
-import { View, Text, StyleSheet, TouchableOpacity, TextInput, Platform } from 'react-native';
+import {
+  View,
+  Text,
+  StyleSheet,
+  TouchableOpacity,
+  TextInput,
+  Platform,
+  DimensionValue,
+} from 'react-native';
 import { Plant } from '../types';
 import { ActivityBar } from './ActivityBar';
 import { useTheme } from '../hooks/useTheme';
@@ -119,8 +127,8 @@ export const PlantRow: React.FC<PlantRowProps> = ({
           const start = Math.min(dragPreview.startMonth, dragPreview.endMonth);
           const end = Math.max(dragPreview.startMonth, dragPreview.endMonth);
           return {
-            left: `${(start / totalMonths) * 100}%`,
-            width: `${((end - start + 1) / totalMonths) * 100}%`,
+            left: `${(start / totalMonths) * 100}%` as DimensionValue,
+            width: `${((end - start + 1) / totalMonths) * 100}%` as DimensionValue,
           };
         })()
       : null;
@@ -156,7 +164,7 @@ export const PlantRow: React.FC<PlantRowProps> = ({
               onMouseEnter: () => handleCellMouseEnter(monthIndex),
               style: [
                 ...baseStyle,
-                { cursor: 'crosshair' } as React.ComponentProps<typeof View>['style'],
+                { cursor: 'crosshair' } as unknown as React.ComponentProps<typeof View>['style'],
               ],
             };
             return <View key={monthIndex} {...(webProps as React.ComponentProps<typeof View>)} />;

--- a/src/components/PlantRowsContainer.tsx
+++ b/src/components/PlantRowsContainer.tsx
@@ -15,6 +15,7 @@ interface PlantRowsContainerProps {
   cellWidth?: number;
   onPressActivity: (plantId: string, activityId: string) => void;
   onPressMonth: (plantId: string, monthIndex: number) => void;
+  onPressMonthRange: (plantId: string, startMonth: number, endMonth: number) => void;
   onFixedScrollOffset?: (offset: number) => void;
   fixedScrollRef?: React.RefObject<ScrollView | null>;
   headerScrollRef?: React.RefObject<ScrollView | null>;
@@ -29,6 +30,7 @@ export const PlantRowsContainer: React.FC<PlantRowsContainerProps> = ({
   cellWidth: cellWidthProp,
   onPressActivity,
   onPressMonth,
+  onPressMonthRange,
   onFixedScrollOffset,
   fixedScrollRef: externalFixedScrollRef,
   headerScrollRef,
@@ -151,6 +153,9 @@ export const PlantRowsContainer: React.FC<PlantRowsContainerProps> = ({
                       plant={visiblePlant}
                       onPressActivity={(activityId) => onPressActivity(plant.id, activityId)}
                       onPressMonth={(monthIndex) => onPressMonth(plant.id, monthIndex)}
+                      onPressMonthRange={(startMonth, endMonth) =>
+                        onPressMonthRange(plant.id, startMonth, endMonth)
+                      }
                       onPressPlant={() => {}}
                       totalMonths={months.length}
                       currentHalfMonth={

--- a/src/screens/CalendarScreen.tsx
+++ b/src/screens/CalendarScreen.tsx
@@ -20,6 +20,7 @@ export const CalendarScreen: React.FC = () => {
   const [selectedPlantId, setSelectedPlantId] = useState<string | null>(null);
   const [selectedActivityId, setSelectedActivityId] = useState<string | null>(null);
   const [selectedMonth, setSelectedMonth] = useState(0);
+  const [selectedEndMonth, setSelectedEndMonth] = useState<number | undefined>(undefined);
   const [activeCategory, setActiveCategory] = useState<CategoryFilter>('all');
   const [zoomLevel, setZoomLevel] = useState(2);
   const fixedScrollRef = useRef<ScrollView>(null);
@@ -77,6 +78,16 @@ export const CalendarScreen: React.FC = () => {
   const handlePressMonth = (plantId: string, monthIndex: number) => {
     setSelectedPlantId(plantId);
     setSelectedMonth(isPortrait ? monthIndex * 4 : monthIndex);
+    setSelectedEndMonth(undefined);
+    setShowAddActivity(true);
+  };
+
+  const handlePressMonthRange = (plantId: string, startIdx: number, endIdx: number) => {
+    setSelectedPlantId(plantId);
+    const startHalf = isPortrait ? startIdx * 4 : startIdx;
+    const endHalf = isPortrait ? Math.min(endIdx * 4 + 3, 23) : endIdx;
+    setSelectedMonth(startHalf);
+    setSelectedEndMonth(endHalf);
     setShowAddActivity(true);
   };
 
@@ -153,6 +164,7 @@ export const CalendarScreen: React.FC = () => {
         cellWidth={cellWidth}
         onPressActivity={handlePressActivity}
         onPressMonth={handlePressMonth}
+        onPressMonthRange={handlePressMonthRange}
         fixedScrollRef={fixedScrollRef}
         headerScrollRef={headerScrollRef}
         loading={loading}
@@ -163,6 +175,7 @@ export const CalendarScreen: React.FC = () => {
           visible={showAddActivity}
           plantName={selectedPlant.name}
           initialMonth={selectedMonth}
+          initialEndMonth={selectedEndMonth}
           onClose={() => setShowAddActivity(false)}
           onAdd={handleAddActivity}
         />

--- a/src/screens/SettingsScreen.tsx
+++ b/src/screens/SettingsScreen.tsx
@@ -12,7 +12,7 @@ export const SettingsScreen: React.FC = () => {
 
   const handleClose = () => {
     setModalVisible(false);
-    router.replace('/(tabs)');
+    router.replace('/');
   };
 
   return (

